### PR TITLE
Remove pack200 support

### LINKED ISSUE #478: completely remove pack200
Remove everything related to pack200

### PR DESCRIPTION
**Pull request by akurtakov**
Pack200 specific plugins, options and etc. should be gone now.
Fixes #478

### CODE REVIEW

**Comment by laeubi** [comment]
@akurtakov can we wait to merge this until the Tycho 1.7 release? Just in case we need to cherry-pick something to not get too much conflicts.

**Comment by akurtakov** [comment]
Sure, I just used the time of quiet week on platform for it :)

**Comment by laeubi** [comment]
> Sure, I just used the time of quiet week on platform for it :)

We probably should have more quiet periods then :-D

**Comment by akurtakov** [comment]
Can I merge this one now?

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -6,7 +6,11 @@ This page describes the noteworthy improvements provided by each release of Ecli
 
 ### Migration guide 2.x -> 3.x
 
-- ...
+#### Pack200
+
+Pack200 technology is obsolete and no longer supported after Java 13. Tycho removed all support for dealing with pack.gz files including pack200 specific plugins, various options in Mojos and support for resolving fetching in core. 
+
+Omit any pack200 specific plugins, options and in any other way dealing with ***.pack.gz** files.
```
Comment (laeubi) [inline RELEASE_NOTES.md:13]: Do you think we can have a list of Mojos and Options that are removed?

Comment (akurtakov) [reply RELEASE_NOTES.md:13]: Sure, will do.

File: tycho-bundles/org.eclipse.tycho.p2.maven.repository/src/main/java/org/eclipse/tycho/repository/local/MirroringArtifactProvider.java [hunk 2]
```diff
@@ -395,8 +361,7 @@ public boolean isFileAlreadyAvailable(IArtifactKey artifactKey) {
         }
         Map<String, String> map = new LinkedHashMap<>(descriptor.getProperties());
         //fix bad metadata in p2...
-        if (ArtifactTransferPolicy.isCanonicalFormat(descriptor)
-                && TychoConstants.PACK200_CLASSIFIER.equals(map.get(TychoConstants.PROP_CLASSIFIER))) {
```
Comment (laeubi) [inline tycho-bundles/org.eclipse.tycho.p2.maven.repository/src/main/java/org/eclipse/tycho/repository/local/MirroringArtifactProvider.java:399]: I think we should keep this here to only fix the metadata if pack200 classifier is found.

**Review by laeubi — APPROVED** [review]
Looks good.

**Pull request merged**